packages/flutter_tools/test/integration.shard/gen_l10n_test.dart

```diff
@@ -125,8 +131,29 @@ void main() {
       '#l10n 46 (Hello two worlds with 102 total citizens)\n'
       '#l10n 47 ([Hello] -World- #123#)\n'
       '#l10n 48 (Flutter\'s amazing!)\n'
-      '#l10n 49 (Flutter is "amazing"!)\n'
+      '#l10n 49 (Flutter\'s amazing, times 1!)\n'
+      '#l10n 50 (Flutter is "amazing"!)\n'
       '#l10n END\n'
     );
+  }
+
+  test('generated l10n classes produce expected localized strings', () async {
+    setUpAndRunGenL10n();
+    final StringBuffer stdout = await listenToStdOut(() {
+      _flutter.run();
+    });
+    expectOutput(stdout);
+  });
+
+  test('generated l10n classes produce expected localized strings with deferred loading', () async {
+    setUpAndRunGenL10n(args: <String>['--use-deferred-loading']);
+    // When loading the localizations with deferred loading, we will not have
+    // the localizations the first build, thus we run the app and reload it once
+    // to get the output.
```

Comment:
My assumption is that we need a way to wait for subsequent builds to asynchronously load the localization files it needs so we can test that the right localized strings are being used. However, this feels like an odd way to test intended behavior since using a hot reload doesn't really simulate the real use case. On the other hand, I'm not exactly sure what a better way of waiting for the deferred loading to occur is here.

Also, I don't remember what platform the integration test simulates, but if it's not web, is this expected behavior?

Reply:
You're right, now with the change I am doing we will only load the library for the web. This means that the test does not need to use this hot reload anymore.